compiler/src/dotty/tools/dotc/transform/TypeTestsCasts.scala

```diff
@@ -96,23 +96,30 @@ trait TypeTestsCasts {
         /** Transform isInstanceOf OrType
          *
          *    expr.isInstanceOf[A | B]  ~~>  expr.isInstanceOf[A] | expr.isInstanceOf[B]
+         *    expr.isInstanceOf[A & B]  ~~>  expr.isInstanceOf[A] & expr.isInstanceOf[B]
```

Comment:
Would you mind moving this commit to a separate PR? It's not related to fixing #2072 and I'd like to get it in soon because https://github.com/lampepfl/dotty/pull/2082 depends on it.